src/coord/src/catalog.rs

```diff
@@ -1351,6 +1360,33 @@ impl Catalog {
         }
     }
 
+    /// Returns the [`Op`]s necessary to enable an index.
+    ///
+    /// # Panics
+    /// Panics if `id` is not the `id` of a [`CatalogItem::Index`].
+    pub fn enable_index_ops(&mut self, id: GlobalId) -> Vec<Op> {
+        let mut update = Vec::with_capacity(1);
+
+        let mut index_entry = self.get_by_id(&id).clone();
```

Comment:
Not that it really matters, but the clone here is unnecessary. I think a structure like this might work better:

```rust
    pub fn enable_index_ops(&mut self, id: GlobalId) -> Vec<Op> {
        match &self.get_by_id(&id).item {
            CatalogItem::Index(index) if index.enabled => vec![], // no-op
            CatalogItem::Index(index) => {
                vec![Op::UpdateItem {
                    id,
                    to_item: CatalogItem::Index(Index {enabled: true, ..index.clone()}),
                }]
            }
            _ => unreachable!("cannot enable non-indexes")
        }
    }
```

Reply:
ah, ty--I figured there was a more idiomatic way to express this but just hadn't figured it out